h2o-core/src/main/java/hex/Model.java

```diff
@@ -1444,81 +1451,107 @@ public boolean testJavaScoring(Frame data, Frame model_predictions, double rel_e
 
       String modelName = JCodeGen.toJavaId(_key.toString());
       boolean preview = false;
-      String java_text = toJava(preview, true);
-      GenModel genmodel;
-      try {
-        Class clz = JCodeGen.compile(modelName,java_text);
-        genmodel = (GenModel)clz.newInstance();
-      } catch (Exception e) {
-        e.printStackTrace();
-        throw H2O.fail("Internal POJO compilation failed",e);
-      }
-
+      GenModel genmodel = null;
       Vec[] dvecs = fr.vecs();
       Vec[] pvecs = model_predictions.vecs();
-
-      double[] features = MemoryManager.malloc8d(genmodel._names.length);
-      double[] predictions = MemoryManager.malloc8d(genmodel.nclasses() + 1);
-
-      // Compare predictions, counting mis-predicts
+      double features[] = null;
       int totalMiss = 0;
       int miss = 0;
-      for (int row=0; row<fr.numRows(); row++) { // For all rows, single-threaded
-
-        // Native Java API
-        for (int col = 0; col < features.length; col++) // Build feature set
-          features[col] = dvecs[col].at(row);
-        genmodel.score0(features, predictions);            // POJO predictions
-        for (int col = 0; col < pvecs.length; col++) { // Compare predictions
-          double d = pvecs[col].at(row);                  // Load internal scoring predictions
-          if (col == 0 && omap != null) d = omap[(int) d];  // map categorical response to scoring domain
-          if (!MathUtils.compare(predictions[col], d, 1e-15, rel_epsilon)) {
-            if (miss++ < 10)
-              System.err.println("Predictions mismatch, row " + row + ", col " + model_predictions._names[col] + ", internal prediction=" + d + ", POJO prediction=" + predictions[col]);
+
```

Comment:
Minor comment: seems complicated - would be better to split into multiple method.